devel/ruff/Portfile

```diff
@@ -30,16 +30,36 @@ maintainers         {gmail.com:herby.gillot @herbygillot} \
                     openmaintainer
 
 checksums           ${distname}${extract.suffix} \
-                    rmd160  5c1d8a1c2275807c107d5738b2f2faa9c106d4e3 \
-                    sha256  cd3212c8e44c0c2c92eba10c44df0ed4073c77b2cff669563b37a56a090c9e74 \
-                    size    4921666
+                    rmd160  8c31d22d81697725425c33769dd9ac27f482fdc9 \
+                    sha256  d82f44f45fc310345cb7d4ce3fc5c9a39556515062002804bb4ae9f6e191c2b2 \
+                    size    4944255
 
 cargo.offline_cmd
 
+# Generate shell completions for supported shells
+post-build {
+    set ruff_bin ${worksrcpath}/target/[cargo.rust_platform]/release/${name}
+    foreach shell {zsh bash fish} {
+        system -W ${worksrcpath} "${ruff_bin} generate-shell-completion ${shell} > ruff.${shell}"
+    }
+}
+
 destroot {
-    xinstall -m 0755 \
+    xinstall -m 755 \
```

Comment:
Why `755` instead of `0755`?

Reply:
I think I changed it to be consistent with the new changes made, but reverting it back is no problem

Reply:
Yeah, let's keep it. Everything else looks good otherwise.